nakama/engine/defold.lua

```diff
@@ -66,9 +66,14 @@ function M.http(config, url_path, query_params, method, post_data, callback)
 	log("HTTP", method, url)
 	log("DATA", post_data)
 	http.request(url, method, function(self, id, result)
-		if result.response then
-			log(result.response)
-			result.response = json.decode(result.response)
+		log(result.response)
+		local ok, decoded = pcall(json.decode, result.response)
+		if not ok then
+			result.response = { error = true, message = "Unable to decode response" }
```

Comment:
The response was not JSON (or it was malformed). Is it safe to assume that it is an error in this case?

Reply:
Yes, it's safe to assume. If there's a bug which will fail the request it could cause a load balancer to return a HTML error response.